src/tests/uat/recommendation_uat.py

```diff
@@ -0,0 +1,423 @@
+
+    Args:
+        recommendation_list (list[Recommendation]): List of possible recommendations to be our candidate.
+
+    Returns:
+        Recommendation: A candidate recommendation for testing. Looks for a recommendation with NEW status by
+                        preference, then REJECTED, and then ACCEPTED. If none is found, returns None.
+    """
+    for initial_status in ['NEW', 'REJECTED', 'ACCEPTED']:
+        rec_candidates = [rec for rec in recommendation_list
+                          if rec.rule_type == 'reputation_override' and rec.workflow_.status == initial_status]
+        if len(rec_candidates) > 0:
+            return rec_candidates[0]
+    return None
+
+
+def exercise_workflow(recommendation, config_data, verbose):
+    """
+    Exercises workflow operations on a recommendation.
+
+    Args:
+        recommendation (Recommendation): The recommendation value under test.
+        config_data (dict): Contains configuration data for the request.
+        verbose (bool): True to print slightly more information.
+
+    Returns:
+        bool: True if exercise was successful, False if it failed.
+    """
+    initial_status = recommendation.workflow_.status
+    result_workflow = validate_status_via_raw(recommendation, config_data)
+    if initial_status != 'NEW':
+        if verbose:
+            print("--- Resetting recommendation")
+        recommendation.reset()
+        result_workflow = validate_status_via_raw(recommendation, config_data) and result_workflow
+    if initial_status != 'ACCEPTED':
+        if verbose:
+            print("--- Accepting recommendation")
+        recommendation.accept()
+        result_workflow = validate_status_via_raw(recommendation, config_data) and result_workflow
+        if verbose:
+            print("--- Resetting recommendation")
+        recommendation.reset()
+        result_workflow = validate_status_via_raw(recommendation, config_data) and result_workflow
+    if initial_status != 'REJECTED':
+        if verbose:
+            print("--- Rejecting recommendation")
+        recommendation.reject()
+        result_workflow = validate_status_via_raw(recommendation, config_data) and result_workflow
+        if verbose:
+            print("--- Resetting recommendation")
+        recommendation.reset()
+        result_workflow = validate_status_via_raw(recommendation, config_data) and result_workflow
+    if initial_status == 'ACCEPTED':
+        if verbose:
+            print("--- Accepting recommendation")
+        recommendation.accept()
+        result_workflow = validate_status_via_raw(recommendation, config_data) and result_workflow
+    elif initial_status == 'REJECTED':
+        if verbose:
+            print("--- Rejecting recommendation")
+        recommendation.reject()
+        result_workflow = validate_status_via_raw(recommendation, config_data) and result_workflow
+    return result_workflow
```

Comment:
Is it possible by running this test to break another one that is depending on the selected recommendation if we don't reset it to its initial state?

Reply:
There aren't any other tests involving recommendations. This is the only one.

Reply:
I mean of another team. For example, if we run this test on test03 during bug bash on the same org as other teams, can we break their tests?

Reply:
Possible, but how likely is that to happen? There's certainly no way to "sync" between two running instances of the test.

Reply:
I agree that it would probably be a rare situation. And I am thinking... even if we use a recommendation that a different simultaneously running test also attempts to use, resetting the state after the test will not do much to help. Cause they won't wait in order. I'm ok to leave it as is.